src/policy/mallocspace/global.rs

```diff
@@ -235,6 +239,32 @@ impl<VM: VMBinding> MallocSpace<VM> {
         }
     }
 
+    // This is called by mutators. We have to make sure this is correct if more than one threads are calling this method for
+    // objects in the same page.
+    fn set_page_mark(&self, start: Address, size: usize) {
+        // Set first page
+        let mut page = start.align_down(BYTES_IN_MALLOC_PAGE);
+
+        // It is important to go to the end of the object, which may span a page boundary
+        while page < start + size {
+            if compare_exchange_set_page_mark(page) {
```

Comment:
I think performance is not the main goal for this plan/policy. We have many places in the policy that will have overhead. Making efforts in optimizing it will not be very rewarding.

Page mark is one byte per page rather than one bit. So we won't have race in the bits when we do compare-exchange. We could still do `fetch_or` if we care about large objects. Alternatively, we could set a large object threshold for the plan, so large objects are allocated into large object space. 

But anyway, to my understanding, we are still uncertain on whether we keep malloc mark sweep or just drop it. I feel spending too much efforts it may not worth it.

Reply:
I agree with Yi here. The only focus on this should be correctness not performance, which is why the original fix in my branch didn't optimize everything possible. I'm not if this is a fruitful endeavour specially since we're switching over to the native-Rust mimalloc very soon as well.

Reply:
OK. That makes sense.  The current approach is still correct.